### ISSUE TITLE
Incorrect call to executable-find when searching for the `fd` or `fdfind` executable in projectile commands

### ISSUE THREAD

**Issue by ashutoshrishi** [hlissner/doom-emacs#3455]
**What did you expect to happen?**

Using projectile commands after updating to the latest develop doesn't work. There is an error thrown:

```
Projectile is initializing cache for <REDACTED>...
apply: Wrong number of arguments: (1 . 1), 2
```

**What actually happened?**

Using the emacs debugger, the faulting commit might be https://github.com/hlissner/doom-emacs/commit/d034bc2004ad6873d80e9cff1ef69b4445137135#diff-021d919f1b2767331792589616c1561fR157

This results in `exectuable-find` being *applied* with 2 arguments: `("fdfind" t)` 

I am not an expert in elisp, but this seems to work:

```elisp
(or (cl-find-if (doom-rpartial #'executable-find)
                                            '("fdfind" "fd"))
                                "fd"))
```
^ just removes the `t` argument to `doom-rpartial` as it should not be *bound* as an argument to `executable-find`.

**Additional details:**

**BACKTRACE**

<details><pre>
Debugger entered--Lisp error: (wrong-number-of-arguments (1 . 1) 2)
  executable-find("fdfind" t)
  apply(executable-find ("fdfind" t))
  (closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args)))("fdfind")
  cl--position(nil ("fdfind" "fd") 0 nil nil)
  cl-position(nil ("fdfind" "fd") :if (closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args))))
  apply(cl-position nil ("fdfind" "fd") (:if (closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args)))))
  cl-find(nil ("fdfind" "fd") :if (closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args))))
  apply(cl-find nil ("fdfind" "fd") :if (closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args))) nil)
  cl-find-if((closure ((args t) (fn . executable-find) t) (&rest pre-args) (apply fn (append pre-args args))) ("fdfind" "fd"))
  (or (cl-find-if (doom-rpartial (function executable-find) t) (quote ("fdfind" "fd"))) "fd")
  (format "%s . -0 -H -E .git --color=never --type file --type symlink --follow" (or (cl-find-if (doom-rpartial (function executable-find) t) (quote ("fdfind" "fd"))) "fd"))
  (concat (format "%s . -0 -H -E .git --color=never --type file --type symlink --follow" (or (cl-find-if (doom-rpartial (function executable-find) t) (quote ("fdfind" "fd"))) "fd")) (let* ((--cl-var-- projectile-globally-ignored-directories) (dir nil) (--cl-var-- "")) (while (consp --cl-var--) (setq dir (car --cl-var--)) (setq --cl-var-- (concat --cl-var-- " -E ")) (setq --cl-var-- (concat --cl-var-- (shell-quote-argument dir))) (setq --cl-var-- (cdr --cl-var--))) --cl-var--) (if IS-WINDOWS " --path-separator=//"))
  (closure (t) (_) (concat (format "%s . -0 -H -E .git --color=never --type file --type symlink --follow" (or (cl-find-if (doom-rpartial (function executable-find) t) (quote ("fdfind" "fd"))) "fd")) (let* ((--cl-var-- projectile-globally-ignored-directories) (dir nil) (--cl-var-- "")) (while (consp --cl-var--) (setq dir (car --cl-var--)) (setq --cl-var-- (concat --cl-var-- " -E ")) (setq --cl-var-- (concat --cl-var-- (shell-quote-argument dir))) (setq --cl-var-- (cdr --cl-var--))) --cl-var--) (if IS-WINDOWS " --path-separator=//")))(git)
  funcall((closure (t) (_) (concat (format "%s . -0 -H -E .git --color=never --type file --type symlink --follow" (or (cl-find-if (doom-rpartial (function executable-find) t) (quote ("fdfind" "fd"))) "fd")) (let* ((--cl-var-- projectile-globally-ignored-directories) (dir nil) (--cl-var-- "")) (while (consp --cl-var--) (setq dir (car --cl-var--)) (setq --cl-var-- (concat --cl-var-- " -E ")) (setq --cl-var-- (concat --cl-var-- (shell-quote-argument dir))) (setq --cl-var-- (cdr --cl-var--))) --cl-var--) (if IS-WINDOWS " --path-separator=//"))) git)
  (if (functionp projectile-generic-command) (funcall projectile-generic-command vcs) projectile-generic-command)
  doom--only-use-generic-command-a(git)
  apply(doom--only-use-generic-command-a git)
  projectile-get-ext-command(git)
  projectile-dir-files-alien("/Users/zeus/projects/pantheon/aphrodite/")
  projectile-project-files("/Users/zeus/projects/pantheon/aphrodite/")
  projectile-current-project-files()
  counsel-projectile-find-file(nil)
  funcall-interactively(counsel-projectile-find-file nil)
  call-interactively(counsel-projectile-find-file)
  (cond ((and projectile-project-root (file-equal-p projectile-project-root default-directory)) (if (doom-project-p default-directory) nil (setq projectile-enable-caching nil)) (call-interactively (if (doom-module-p :completion (quote ivy)) (function counsel-projectile-find-file) (function projectile-find-file)))) ((fboundp (quote counsel-file-jump)) (call-interactively (function counsel-file-jump))) ((project-current) (project-find-file-in nil (list default-directory) nil)) ((fboundp (quote helm-find-files)) (call-interactively (function helm-find-files))) ((call-interactively (function find-file))))
  (let* ((default-directory (file-truename (expand-file-name dir))) (projectile-project-root (doom-project-root default-directory)) (projectile-enable-caching projectile-enable-caching)) (cond ((and projectile-project-root (file-equal-p projectile-project-root default-directory)) (if (doom-project-p default-directory) nil (setq projectile-enable-caching nil)) (call-interactively (if (doom-module-p :completion (quote ivy)) (function counsel-projectile-find-file) (function projectile-find-file)))) ((fboundp (quote counsel-file-jump)) (call-interactively (function counsel-file-jump))) ((project-current) (project-find-file-in nil (list default-directory) nil)) ((fboundp (quote helm-find-files)) (call-interactively (function helm-find-files))) ((call-interactively (function find-file)))))
  doom-project-find-file("~/projects/pantheon/aphrodite/")
  funcall(doom-project-find-file "~/projects/pantheon/aphrodite/")
  (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir))
  (if (and (not (null +workspaces-on-switch-project-behavior)) (or (eq +workspaces-on-switch-project-behavior t) (equal (safe-persp-name (get-current-persp)) persp-nil-name) (+workspace-buffer-list))) (let* ((persp (let ((project-name (doom-project-name +workspaces--project-dir))) (or (+workspace-get project-name t) (+workspace-new project-name)))) (new-name (progn (or (and (memq ... cl-struct-perspective-tags) t) (signal (quote wrong-type-argument) (list ... persp))) (aref persp 1)))) (+workspace-switch new-name) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)) (+workspace-message (format "Switched to '%s' in new workspace" new-name) (quote success))) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (message "Switched to '%s'" (doom-project-name +workspaces--project-dir))) (condition-case err (+workspace-rename (+workspace-current-name) (doom-project-name +workspaces--project-dir)) ((debug error) (message "Workspace error: %s" err) nil)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)))
  (unwind-protect (if (and (not (null +workspaces-on-switch-project-behavior)) (or (eq +workspaces-on-switch-project-behavior t) (equal (safe-persp-name (get-current-persp)) persp-nil-name) (+workspace-buffer-list))) (let* ((persp (let ((project-name ...)) (or (+workspace-get project-name t) (+workspace-new project-name)))) (new-name (progn (or (and ... t) (signal ... ...)) (aref persp 1)))) (+workspace-switch new-name) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)) (+workspace-message (format "Switched to '%s' in new workspace" new-name) (quote success))) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (message "Switched to '%s'" (doom-project-name +workspaces--project-dir))) (condition-case err (+workspace-rename (+workspace-current-name) (doom-project-name +workspaces--project-dir)) ((debug error) (message "Workspace error: %s" err) nil)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir))) (run-hooks (quote projectile-after-switch-project-hook)) (setq +workspaces--project-dir nil))
  (progn (if projectile-before-switch-project-hook (progn (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (run-hooks ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))))) (unwind-protect (if (and (not (null +workspaces-on-switch-project-behavior)) (or (eq +workspaces-on-switch-project-behavior t) (equal (safe-persp-name (get-current-persp)) persp-nil-name) (+workspace-buffer-list))) (let* ((persp (let (...) (or ... ...))) (new-name (progn (or ... ...) (aref persp 1)))) (+workspace-switch new-name) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)) (+workspace-message (format "Switched to '%s' in new workspace" new-name) (quote success))) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (message "Switched to '%s'" (doom-project-name +workspaces--project-dir))) (condition-case err (+workspace-rename (+workspace-current-name) (doom-project-name +workspaces--project-dir)) ((debug error) (message "Workspace error: %s" err) nil)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir))) (run-hooks (quote projectile-after-switch-project-hook)) (setq +workspaces--project-dir nil)))
  (if (and persp-mode +workspaces--project-dir) (progn (if projectile-before-switch-project-hook (progn (let ((temp-buffer (generate-new-buffer " *temp*"))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn ... ... ...) (and ... ...)))))) (unwind-protect (if (and (not (null +workspaces-on-switch-project-behavior)) (or (eq +workspaces-on-switch-project-behavior t) (equal (safe-persp-name ...) persp-nil-name) (+workspace-buffer-list))) (let* ((persp (let ... ...)) (new-name (progn ... ...))) (+workspace-switch new-name) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)) (+workspace-message (format "Switched to '%s' in new workspace" new-name) (quote success))) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (message "Switched to '%s'" (doom-project-name +workspaces--project-dir))) (condition-case err (+workspace-rename (+workspace-current-name) (doom-project-name +workspaces--project-dir)) ((debug error) (message "Workspace error: %s" err) nil)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir))) (run-hooks (quote projectile-after-switch-project-hook)) (setq +workspaces--project-dir nil))))
  (let (projectile-project-root) (if (and persp-mode +workspaces--project-dir) (progn (if projectile-before-switch-project-hook (progn (let ((temp-buffer ...)) (save-current-buffer (set-buffer temp-buffer) (unwind-protect ... ...))))) (unwind-protect (if (and (not (null +workspaces-on-switch-project-behavior)) (or (eq +workspaces-on-switch-project-behavior t) (equal ... persp-nil-name) (+workspace-buffer-list))) (let* ((persp ...) (new-name ...)) (+workspace-switch new-name) (save-current-buffer (set-buffer ...) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir)) (+workspace-message (format "Switched to '%s' in new workspace" new-name) (quote success))) (save-current-buffer (set-buffer (doom-fallback-buffer)) (setq default-directory +workspaces--project-dir) (hack-dir-local-variables-non-file-buffer) (message "Switched to '%s'" (doom-project-name +workspaces--project-dir))) (condition-case err (+workspace-rename (+workspace-current-name) (doom-project-name +workspaces--project-dir)) ((debug error) (message "Workspace error: %s" err) nil)) (if current-prefix-arg nil (funcall +workspaces-switch-project-function +workspaces--project-dir))) (run-hooks (quote projectile-after-switch-project-hook)) (setq +workspaces--project-dir nil)))))
  +workspaces-switch-to-project-h("~/projects/pantheon/aphrodite/")
  ivy-call()
  ivy-read("[-] Switch to project: " ("~/projects/pantheon/aphrodite/" "~/projects/pantheon/dummy/") :preselect nil :action (1 ("o" +workspaces-switch-to-project-h "open project in new workspace") ("O" counsel-projectile-switch-project-action "jump to a project buffer or file") ("f" counsel-projectile-switch-project-action-find-file "jump to a project file") ("d" counsel-projectile-switch-project-action-find-dir "jump to a project directory") ("b" counsel-projectile-switch-project-action-switch-to-buffer "jump to a project buffer") ("m" counsel-projectile-switch-project-action-find-file-manually "find file manually from project root") ("w" counsel-projectile-switch-project-action-save-all-buffers "save all project buffers") ("k" counsel-projectile-switch-project-action-kill-buffers "kill all project buffers") ("r" counsel-projectile-switch-project-action-remove-known-project "remove project from known projects") ("c" counsel-projectile-switch-project-action-compile "run project compilation command") ("C" counsel-projectile-switch-project-action-configure "run project configure command") ("e" counsel-projectile-switch-project-action-edit-dir-locals "edit project dir-locals") ("v" counsel-projectile-switch-project-action-vc "open project in vc-dir / magit / monky") ("s" (lambda (project) (let ((projectile-switch-project-action (lambda nil ...))) (counsel-projectile-switch-project-by-name project))) "search project") ("xs" counsel-projectile-switch-project-action-run-shell "invoke shell from project root") ("xe" counsel-projectile-switch-project-action-run-eshell "invoke eshell from project root") ("xt" counsel-projectile-switch-project-action-run-term "invoke term from project root") ("X" counsel-projectile-switch-project-action-org-capture "org-capture into project")) :require-match t :sort nil :caller counsel-projectile-switch-project)
  counsel-projectile-switch-project()
  funcall-interactively(counsel-projectile-switch-project)
  call-interactively(counsel-projectile-switch-project nil nil)
  command-execute(counsel-projectile-switch-project)

</pre></details>

**System information:**

<details>

```
emacs   version    26.3
        features   NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS LCMS2
        build      Jun 15, 2020
        buildopts  (--with-mac --enable-mac-app=/Users/travis/build/railwaycat/homebrew-emacsmacport/build-scripts/emacs-source/tmproot --prefix=/Users/travis/build/railwaycat/homebrew-emacsmacport/build-scripts/emacs-source/tmproot --enable-mac-self-contained --with-modules)
        windowsys  mac
        daemonp    server-running
doom    version    2.0.9
        build      grafted, HEAD -> develop, origin/develop, origin/HEAD 0443eeb 2020-06-24 20:09:17 -0400
        dir        ~/.doom.d/
system  type       darwin
        config     x86_64-apple-darwin18.7.0
        shell      /usr/local/bin/zsh
        uname      Darwin 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
        path       (~/.local/src/elixir-ls/release ~/.local/bin ~/.cabal/bin ~/.ghcup/bin ~/.opam/4.02.3/bin ~/.pyenv/shims ~/.rbenv/shims ~/.nvm/versions/node/v12.11.0/bin ~/.local/bin ~/.cargo/bin ~/.local/bin /usr/local/bin /usr/bin /bin /usr/sbin /sbin /usr/local/go/bin /opt/X11/bin /Applications/Postgres.app/Contents/Versions/latest/bin node_modules/.bin ~/go/bin ~/.emacs.d/bin node_modules/.bin ~/go/bin ~/.emacs.d/bin node_modules/.bin ~/go/bin ~/.emacs.d/bin /Applications/Emacs.app/Contents/MacOS/libexec/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline neotree ophints (popup +defaults) vc-gutter vi-tilde-fringe workspaces :editor file-templates fold (format +onsave) multiple-cursors snippets :emacs dired electric undo vc :checkers syntax :tools (eval +overlay) lookup lsp macos magit terraform :lang data (elixir +lsp) emacs-lisp (haskell +dante) json (javascript +lsp) markdown org rust sh (web +lsp) yaml :config (default +bindings +smartparens))
        packages   ((ace-jump-mode) (add-node-modules-path) (material-theme) (prettier-js))
        unpin      (n/a)
        elpa       (n/a)
```
</details>

**Comment by benmezger** [comment]
This saved me a bunch of time 😆

**Comment by hlissner** [comment]
Ah, just realized `executable-find`'s second argument was added in Emacs 27, so 26 users will see this error. Should be fixed as of 00ebbaf. Thanks for bringing it to my attention!

**Comment by ashutoshrishi** [comment]
@hlissner Ahh I thought it must be a Emacs version library API difference. Thanks!
